examples/rdkafka_performance.c

```diff
@@ -1400,11 +1400,15 @@ int main (int argc, char **argv) {
 			cnt.msgs++;
 			cnt.bytes += msgsize;
 
-                        if (rate_sleep)
-                                do_sleep(rate_sleep);
-
 			/* Must poll to handle delivery reports */
-			rd_kafka_poll(rk, 0);
+			if (rate_sleep) {
+				rd_ts_t next = rd_clock() + (rd_ts_t) rate_sleep;
+				do {
+					rd_kafka_poll(rk, RD_MIN(100, rate_sleep));
```

Comment:
rd_kafka_poll()'s timeout is in milliseconds while rate_sleep is in microseconds.

Reply:
I suggest calculating the remaining time (of rate_sleep) in each loop and using that for timeout, which removes the need for RD_MIN()

Reply:
Done -- but now I ended up with an RD_MAX to avoid polling with a negative timeout :)